Tools/TestWebKitAPI/Tests/WebKitCocoa/WKWebExtensionAPISidebar.mm

```diff
@@ -714,6 +719,204 @@
     [manager loadAndRun];
 }
 
+TEST_F(WKWebExtensionAPISidebar, SidebarActionOpenFailsWithoutUserGesture)
+{
+    auto *script = @[
+        @"await browser.sidebarAction.open()",
+        @"    .then(() => browser.test.notifyFail('sidebarAction.open() did not throw without user gesture'))",
+        @"    .catch(() => browser.test.notifyPass())",
```

Comment:
You don't `await` with `.then()` or `.catch()`. You would use `try { } catch(e) { }` like a normal sync call.

Instead, use:

```suggestion
        @"browser.test.assertThrows(() => browser.sidebarAction.open())",
        @"browser.test.notifyPass()",
```

The assert will fail the test, even if the pass is hit later.

Reply:
Don't `.then` and `.catch` themselves return promises chained against the initial promise? My intention here is to await the final promise returned by `.catch`.  Regarding `assertThrows` -- that's actually what I tried to use initially. I found that the error returned from `open()` in this scenario only surfaces if the promise returned by `open()` is awaited, so:
```
browser.test.assertThrows(() => browser.sidebarAction.open())
```
 will fail even if the promise is correctly rejected. Unfortunately, we can't do this:
```
browser.test.assertThrows(() => await browser.sidebarAction.open())
```
since it's a syntax error to await in a non-async closure. I also tried this:
```
browser.test.assertThrows(async () => await browser.sidebarAction.open())
```
which doesn't work since `assertThrows` will not await the promise returned by the async closure. So far as I can tell, I don't think we can use `assertThrows` here since the promise is rejected, rather than an exception being directly thrown. This is how I came to the `.then() ... .catch()` chaining. For what it's worth, I played with it experimentally a bit and found that it correctly passes/fails the test based on whether the promise resolves or is rejected.

Reply:
Ah yes! Makes sense. However, the gesture error should be a thrown exception from the web process and not an async error. So `assertThrows` would work then.

Reply:
Fair enough! I forgot about that incoming change as I was writing all this out haha